packages/state-transition/src/cache/epochCache.ts

```diff
@@ -838,15 +1042,53 @@ export class EpochCache {
   }
 
   effectiveBalanceIncrementsSet(index: number, effectiveBalance: number): void {
-    if (index >= this.effectiveBalanceIncrements.length) {
-      // Clone and extend effectiveBalanceIncrements
+    if (this.isAfterElectra()) {
+      // TODO: electra
+      // getting length and setting getEffectiveBalanceIncrementsByteLen is not fork safe
+      // so each time we add an index, we should new the Uint8Array to keep it forksafe
+      // one simple optimization could be to increment the length once per block rather
+      // on each add/set
+      //
+      // there could still be some unused length remaining from the prev ELECTRA padding
+      const newLength =
+        index >= this.effectiveBalanceIncrements.length ? index + 1 : this.effectiveBalanceIncrements.length;
       const effectiveBalanceIncrements = this.effectiveBalanceIncrements;
-      this.effectiveBalanceIncrements = new Uint8Array(getEffectiveBalanceIncrementsByteLen(index + 1));
+      this.effectiveBalanceIncrements = new Uint16Array(newLength);
       this.effectiveBalanceIncrements.set(effectiveBalanceIncrements, 0);
+    } else {
+      if (index >= this.effectiveBalanceIncrements.length) {
+        // Clone and extend effectiveBalanceIncrements
+        const effectiveBalanceIncrements = this.effectiveBalanceIncrements;
+        this.effectiveBalanceIncrements = new Uint16Array(getEffectiveBalanceIncrementsByteLen(index + 1));
+        this.effectiveBalanceIncrements.set(effectiveBalanceIncrements, 0);
+      }
     }
 
     this.effectiveBalanceIncrements[index] = Math.floor(effectiveBalance / EFFECTIVE_BALANCE_INCREMENT);
   }
+
+  isAfterElectra(): boolean {
```

Comment:
consider renaming to `isPostElectra` to match `ForkPostElectra`

Reply:
this should probably be private

Reply:
> this should probably be private

I think `isPostElectra()` provides a convenient way for eth1 deposit tracker to determine when to stop polling data
https://github.com/ChainSafe/lodestar/blob/3f4b2cb9c37f3032bcfc3d947aee5d22b3f03a7a/packages/beacon-node/src/eth1/eth1DepositDataTracker.ts#L131